src/plans/conjugate_gradient_plan.jl

```diff
@@ -77,23 +78,25 @@ mutable struct ConjugateGradientDescentState{
     δ::T
     β::F
     coefficient::TCoeff
+    restart_condition::TRC
     stepsize::TStepsize
     stop::TStop
     retraction_method::TRetr
     vector_transport_method::VTM
-    function ConjugateGradientDescentState{P,T}(
+    function ConjugateGradientDescentState{P,T,TsC,TStep,TRC,TRetr,VTM}(
```

Comment:
```suggestion
    function ConjugateGradientDescentState(
```
I think it does not like (nor need) these types here